content/deploy/restore/keycloak.md

```diff
@@ -0,0 +1,103 @@
+---
+title: 'Restore Keycloak'
+date: '2024-01-08T13:26:23-05:00'
+categories: ["how-to"]
+description: How to restore a Keycloak backup on Rhize
+weight: 300
+menu:
+  main:
+    parent: restore
+    identifier:
+---
+
+This guide shows you how to restore Keycloak in your Rhize environment.
+
+| Restoring Keycloak to an existing running instance will involve downtime
+
+Expected downtime for an existing running instance depends on the cluster performance sizing, backup size and bandwidth speed to the kubernetes cluster. Downtime is typically under a minute.
+
+## Prerequisites
+
+Before you start, ensure you have the following:
+
+- [`kubectl`](https://kubernetes.io/docs/tasks/tools/)
+- A [Keycloak backup]({{< relref "../backup/keycloak" >}})
+
+## Steps
+
+1. Confirm the cluster and namespace are correct:
+
+    {{% param "k8s_cluster_ns" %}}
+
+1. Retrieve the Keycloak user password using the following command, replacing <namespace> with your namespace:
+
+  ```bash
+  kubectl get secret keycloak-<namespace>-postgresql -o jsonpath="{.data.postgres-password}" | base64 --decode
+  ```
+
+1. Extract your backup file 
+
+  ```bash
+  gzip -d keycloak-postgres-backup-YYYYMMDDTHHMMAA.sql
+  ```
+
+1. Scale down the replicas of Keycloak to 0 to prevent new records being created whilst the restore is in progress. Keycloak will be unavailable after this command
```

Comment:
```suggestion
1. Scale down the replicas of Keycloak to 0 to prevent new records from being created while the backup is restored. Keycloak will be unavailable after this command.
```